scipy/integrate/tests/test__quad_vec.py

```diff
@@ -123,3 +144,46 @@ def f_inf(x):
 
     res, err, info = quad_vec(f_inf, 0, 1, full_output=True)
     assert info.status == 3
+
+
+
+def test_points():
+    # Check that initial interval splitting is done according to
+    # `points`, by checking that consecutive sets of 15 point (for
+    # gk15) function evaluations lie between `points`
+
+    def check_quad_points(a, b, points):
+        np.random.seed(1234)
+
+        quadrature_points = 15
+        interval_sets = []
+        count = 0
+
+        def f(x):
+            nonlocal count
+
+            if count % quadrature_points == 0:
+                interval_sets.append(set())
+
+            count += 1
+
+            interval_sets[-1].add(float(x))
+            return np.random.rand()
+
+        quad_vec(f, a, b, points=points, quadrature='gk15', limit=0)
+
+        # Check that all point sets lie in a single `points` interval
+        for p in interval_sets:
+            if not p:
+                continue
+
+            j = np.searchsorted(sorted(points), tuple(p))
+            assert np.all(j == j[0])
+
+    points = (0, 0.25, 0.5, 0.75, 1.0)
+    points += tuple(-x for x in points)
+
+    check_quad_points(0, 1, points=points)
+    check_quad_points(0, np.inf, points=points)
+    check_quad_points(-np.inf, 0, points=points)
+    check_quad_points(-np.inf, np.inf, points=points)
```

Comment:
Any good reason to call `check_quad_points` 4 times inside `test_points()` instead of removing the function nesting & just parametrizing `a` & `b` on a test version of `check_quad_points()` directly?